spec/controllers/idv/in_person/ssn_controller_spec.rb

```diff
@@ -0,0 +1,169 @@
+require 'rails_helper'
+
+RSpec.describe Idv::InPerson::SsnController do
+  include IdvHelper
+
+  let(:flow_session) do
+    { 'document_capture_session_uuid' => 'fd14e181-6fb1-4cdc-92e0-ef66dad0df4e',
+      'pii_from_user' => Idp::Constants::MOCK_IDV_APPLICANT_SAME_ADDRESS_AS_ID_WITH_NO_SSN.dup,
+      :threatmetrix_session_id => 'c90ae7a5-6629-4e77-b97c-f1987c2df7d0',
+      :flow_path => 'standard' }
+  end
+
+  let(:ssn) { Idp::Constants::MOCK_IDV_APPLICANT_WITH_SSN[:ssn] }
+
+  let(:user) { create(:user) }
+
+  before do
+    stub_sign_in(user)
+    subject.user_session['idv/in_person'] = flow_session
+    stub_analytics
+    stub_attempts_tracker
+    allow(@analytics).to receive(:track_event)
+  end
+
+  describe 'before_actions' do
+    it 'includes authentication before_action' do
+      expect(subject).to have_actions(
+        :before,
+        :confirm_two_factor_authenticated,
+      )
+    end
+
+    it 'checks that feature flag is enabled' do
+      expect(subject).to have_actions(
+        :before,
+        :renders_404_if_in_person_ssn_info_controller_enabled_flag_not_set,
+      )
+    end
+
+    it 'includes outage before_action' do
+      expect(subject).to have_actions(
+        :before,
+        :check_for_outage,
+      )
+    end
+
+    it 'checks that the previous step is complete' do
+      expect(subject).to have_actions(
+        :before,
+        :confirm_in_person_address_step_complete,
+      )
```

Comment:
This test seems insufficient to me because the `confirm_in_person_address_step_complete` method is never tested as a unit. So, for example, I can replace `confirm_in_person_address_step_complete` with:

```
def confirm_in_person_address_step_complete
  return
end
```

and nothing breaks. Would prefer to see that method be tested in some way. I think the lowest effort way would be to add a new context, like

```
context('#confirm_in_person_address_step_complete`) do
    it('redirects if the user hasn't completed the address page') do
        // ...
     end
end
```

Reply:
@sheldon-b  Great catch! I updated this test as you suggested. I also added additional tests for the renders_404 before action. I deleted one that was copied from another controller (that we are not doing). See commit fc67cab370b308e597c86f44e09e41f49885d63a